hack/vf-netns-switcher.sh

```diff
@@ -0,0 +1,369 @@
+#!/bin/bash
+
+conf_file=""
+
+declare -a netnses
+
+declare -A pfs
+declare -A pcis
+declare -A pf_port_names
+declare -A pf_switch_ids
+
+TIMEOUT="${TIMEOUT:-2}"
+POLL_INTERVAL="${POLL_INTERVAL:-1}"
+
+while test $# -gt 0; do
+  case "$1" in
+
+   --netns | -n)
+      input=$2
+      local_netns=$(cut -s -d ':' -f 1 <<< $input)
+      local_pfs=$(cut -s -d ':' -f 2 <<< $input)
+      input=""
+
+      if [[ -z "$local_netns" ]];then
+          echo "Error: flag --netns specified but netns is empty, please \
+provide it in the form --netns <netns>:<pf1>,<pf2> !"
+          echo "Exiting!"
+          exit 1
+      fi
+
+      if [[ -z "$local_pfs" ]];then
+          echo "Error: flag --netns specified but pfs is empty, please \
+provide it in the form --netns <netns>:<pf1>,<pf2> !"
+          echo "Exiting!"
+          exit 1
+      fi
+
+      netnses+=("$local_netns")
+
+      pfs["$local_netns"]="$(tr , " " <<< $local_pfs)"
+
+      local_netns=""
+      local_pfs=""
+
+      shift
+      shift
+      ;;
+
+   --conf-file | -c)
+      conf_file=$2
+      if [[ ! -f "$conf_file" ]];then
+          echo "Error: flag --conf-file specified but file $conf_file \
+not found!"
+          exit 1
+      fi
+
+      shift
+      shift
+      ;;
+
+   --help | -h)
+      echo "
+vf-netns-switcher.sh --netns <netns>:<pf1>,<pf2> [--conf-file <>]:
+
+	--netns | -n		The netns and its interfaces to switch the interfaces PFs and VFs to. \
+It must be of the form <netns>:<pf1>,<pf2>. This flag can be repeated to specify more netnses.
+
+	--conf-file | -c	A file to read confs from, this will override cli flags. Conf file should be of the form:
+				[
+				  {
+				    "netns": <netns1>,
+				    "pfs": [
+				      "pf1",
+				      "pf2"
+				    ]
+				  },
+				  {
+				    "netns": <netns2>,
+				    "pfs": [
+				      "pf3",
+				      "pf4"
+				    ]
+				  }
+				]
+
+"
+      exit 0
+      ;;
+
+   *)
+      echo "Error: invalid option: $1!"
+      echo "Exiting..."
+      exit 1
+  esac
+done
+
+get_pcis_from_pfs(){
+    local worker_netns="${1}"
+    shift
+    local interfaces="$@"
+    for interface in $interfaces; do
+        pcis["$interface"]="$(get_pci_from_net_name "$interface" "$worker_netns")"
+    done
+}
+
+get_pci_from_net_name(){
+    local interface_name=$1
+    local worker_netns="${2:-$netns}"
+
+    if [[ -z "$(ip l show $interface_name)" ]];then
+        if [[ -n "$(docker exec -t ${worker_netns} ip l show $interface_name)" ]];then
+            ip netns exec ${worker_netns} bash -c "basename \$(readlink /sys/class/net/${interface_name}/device)"
+            return 0
+        fi
+        echo ""
+        return 1
+    fi
+    basename $(readlink /sys/class/net/${interface_name}/device)
```

Comment:
nit: add `return 0` for consistency ?

although return value is not being used IIUC so you could just use plain `return` in all cases here